pkg/vm/stackitem/serialization.go

```diff
@@ -112,8 +112,9 @@ func (w *serContext) serialize(item Item) error {
 		}
 	case *BigInteger:
 		w.data = append(w.data, byte(IntegerT))
-		data := bigint.ToBytes(t.Value().(*big.Int))
-		w.appendVarUint(uint64(len(data)))
+		v := t.Value().(*big.Int)
+		w.appendVarUint(uint64(bigint.Size(v)))
+		data := bigint.ToPreallocatedBytes(v, w.data[len(w.data):])
```

Comment:
Isn't `w.data[len(w.data):]` always an empty array?

Reply:
It is, but it can have non-zero capacity.

Reply:
I mean, always with zero capacity. But you're right, the capacity still can be non-zero.